templates/generic.php

```diff
@@ -312,6 +364,260 @@
+                    <p>
+                        <strong>Sello de pago:</strong>
+                        <span><?=$this->e($pago20['SelloPago'])?></span>
+                    </p>
+                <?php endif; ?>
+                <?php
+                $doctoRelacionados = $pago20->searchNodes('pago20:DoctoRelacionado');
+                if ($doctoRelacionados->count() > 0) :
+                    ?>
+                    <p style="margin: 10px 0 5px 0;">
+                        <strong>Documentos relacionados</strong>
+                    </p>
+                    <?php foreach ($doctoRelacionados as $doctoRelacionado) : ?>
+                    <p style="margin-bottom: 10px;">
+                        <strong>Id Documento: </strong><span><?=$this->e($doctoRelacionado['IdDocumento'])?></span>
+                        <strong>Serie: </strong><span><?=$this->e($doctoRelacionado['Serie'])?></span>
+                        <strong>Folio: </strong><span><?=$this->e($doctoRelacionado['Folio'])?></span>
+                        <strong>Moneda DR: </strong><span><?=$this->e($doctoRelacionado['MonedaDR'])?></span>
+                        <strong>Equivalencia DR: </strong>
+                            <span><?=$this->e($doctoRelacionado['EquivalenciaDR'])?></span>
+                        <strong>Tipo de cambio DR: </strong>
+                            <span><?=$this->e($doctoRelacionado['TipoCambioDR'])?></span>
+                        <strong>Método de pago DR: </strong>
+                            <span><?=$this->e($doctoRelacionado['MetodoDePagoDR'])?></span>
+                        <strong>Número parcialidad: </strong>
+                            <span><?=$this->e($doctoRelacionado['NumParcialidad'])?></span>
+                        <strong>Importe pagado: </strong><span><?=$this->e($doctoRelacionado['ImpPagado'])?></span>
+                        <strong>Importe saldo insoluto: </strong>
+                            <span><?=$this->e($doctoRelacionado['ImpSaldoInsoluto'])?></span>
+                        <strong>Importe saldo anterior: </strong>
+                            <span><?=$this->e($doctoRelacionado['ImpSaldoAnt'])?></span>
+                        <strong>Objeto Imp DR: </strong>
+                            <span><?=$catalagos->catObjetoImp($doctoRelacionado['ObjetoImpDR'])?></span>
+                    </p>
+                        <?php
+                        $impuestos = $doctoRelacionado->searchNode('pago20:ImpuestosDR');
+                        if (null !== $impuestos) : ?>
+                        <p style="margin: 0px 10px 5px 0;">
+                            <strong>Impuestos Docto Relacionado</strong>
+                        </p>
+                            <?php
+                            $traslados = $impuestos->searchNodes('pago20:RetencionesDR', 'pago20:RetencionDR');
+                            $retenciones = $impuestos->searchNodes('pago20:TrasladosDR', 'pago20:TrasladoDR');
```

Comment:
Variables are mixed up, causing issues when representing those values:
```suggestion
                            $retenciones = $impuestos->searchNodes('pago20:RetencionesDR', 'pago20:RetencionDR');
                            $traslados = $impuestos->searchNodes('pago20:TrasladosDR', 'pago20:TrasladoDR');
```